app/utils/auth0_parser.rb

```diff
@@ -0,0 +1,18 @@
+class Auth0Parser
+  def self.authenticate_request(request)
+    auth_token(request)
+  end
+
+  def self.http_token(request)
+    request.headers['Authorization'].split(' ').last
+  end
+
+  def self.auth_token(request)
+    JsonWebToken.verify(http_token(request))
+  end
+
+  def self.auth0_id(request)
+    authenticate_request(request)
```

Comment:
This doesn't seem necessary. `authenticate_request` only calls `auth_token`, which you're already calling on the next line.